### PR TITLE
Test GCC 13 for Amiga builds

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Packaging/amiga/prep.sh [hunk 1]
```diff
@@ -13,6 +13,8 @@ export M68K_COMMON="-s -ffast-math -fomit-frame-pointer -fbbb=-"
 export M68K_CFLAGS="${M68K_CPU_FPU} ${M68K_COMMON}"
 export M68K_CXXFLAGS="${M68K_CPU_FPU} ${M68K_COMMON}"
 
+ln -s /opt/m68k-amigaos-gcc10 /opt/m68k-amigaos
+
```
Comment (AJenbo) [inline Packaging/amiga/prep.sh:17]:
```suggestion
```

**Comment by AJenbo** [comment]
```
x
_deps/libfmt-build/libfmt.a(format.cc.obj): duplicate section `.text._ZN3fmt2v96detail15write_codepointILm2EcNS0_8appenderEEET1_S4_cm' has different contents
_deps/libfmt-build/libfmt.a(format.cc.obj): duplicate section `.text._ZN3fmt2v96detail15write_codepointILm2EcNS0_8appenderEEET1_S4_cm' has different contents
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/init.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_17LoadMPQERKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS7_EESt17basic_string_viewIcS5_E+0x1cc): undefined reference to `devilution::MpqArchive::Open(char const*, int&)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/init.cpp.obj: in function `_GLOBAL__sub_D__ZN10devilution8gbActiveE':
Source/CMakeFiles/libdevilutionx.dir/init.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_17LoadMPQERKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS7_EESt17basic_string_viewIcS5_E+0x2f8): undefined reference to `devilution::MpqArchive::ErrorMessage(int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/menu.cpp.obj:(.text._ZN10devilution27mainmenu_select_hero_dialogEPNS_8GameDataE+0xd6): undefined reference to `devilution::UiSelHeroMultDialog(bool (*)(bool (*)(devilution::_uiheroinfo*)), bool (*)(devilution::_uiheroinfo*), bool (*)(devilution::_uiheroinfo*), void (*)(unsigned int, devilution::_uidefaultstats*), devilution::_selhero_selections*, unsigned int*)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/menu.cpp.obj:(.text._ZN10devilution27mainmenu_select_hero_dialogEPNS_8GameDataE+0x1b2): undefined reference to `devilution::UiSelHeroMultDialog(bool (*)(bool (*)(devilution::_uiheroinfo*)), bool (*)(devilution::_uiheroinfo*), bool (*)(devilution::_uiheroinfo*), void (*)(unsigned int, devilution::_uidefaultstats*), devilution::_selhero_selections*, unsigned int*)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/menu.cpp.obj:(.text._ZN10devilution27mainmenu_select_hero_dialogEPNS_8GameDataE+0x64): undefined reference to `devilution::UiSelHeroSingDialog(bool (*)(bool (*)(devilution::_uiheroinfo*)), bool (*)(devilution::_uiheroinfo*), bool (*)(devilution::_uiheroinfo*), void (*)(unsigned int, devilution::_uidefaultstats*), devilution::_selhero_selections*, unsigned int*, devilution::_difficulty*)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/menu.cpp.obj:(.text._ZN10devilution27mainmenu_select_hero_dialogEPNS_8GameDataE+0x15a): undefined reference to `devilution::UiSelHeroSingDialog(bool (*)(bool (*)(devilution::_uiheroinfo*)), bool (*)(devilution::_uiheroinfo*), bool (*)(devilution::_uiheroinfo*), void (*)(unsigned int, devilution::_uidefaultstats*), devilution::_selhero_selections*, unsigned int*, devilution::_difficulty*)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/menu.cpp.obj:(.text._ZN10devilution27mainmenu_select_hero_dialogEPNS_8GameDataE+0xec): undefined reference to `devilution::SErrSetLastError(unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/missiles.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_121UpdateMissileVelocityERNS_7MissileENS_7PointOfIiEEi+0x4e): undefined reference to `hypot'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/missiles.cpp.obj: in function `_GLOBAL__sub_D__ZN10devilution8MissilesB5cxx11E':
Source/CMakeFiles/libdevilutionx.dir/missiles.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_121UpdateMissileVelocityERNS_7MissileENS_7PointOfIiEEi+0xe8): undefined reference to `hypot'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/msg.cpp.obj:(.text._ZN10devilution16NetSendCmdStringEmPKc+0x4e): undefined reference to `devilution::multi_send_msg_packet(unsigned long, std::byte const*, unsigned long)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution12NetSendHiPriEiPKSt4bytem.part.0+0x44): undefined reference to `devilution::sync_all_monsters(std::byte*, unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution12NetSendHiPriEiPKSt4bytem.part.0+0x98): undefined reference to `devilution::sync_all_monsters(std::byte*, unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution18multi_handle_deltaEv+0x5c): undefined reference to `devilution::nthread_send_and_recv_turn(unsigned int, int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj: in function `_GLOBAL__sub_D__ZN10devilution23gbSomebodyWonGameKludgeE':
Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution18multi_handle_deltaEv+0x1c2): undefined reference to `devilution::SNetDropPlayer(int, unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution29multi_process_network_packetsEv+0x9c): undefined reference to `devilution::SNetReceiveMessage(unsigned char*, void**, unsigned int*)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj: in function `_GLOBAL__sub_D__ZN10devilution23gbSomebodyWonGameKludgeE':
Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution29multi_process_network_packetsEv+0x39c): undefined reference to `devilution::SNetDropPlayer(int, unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution7NetInitEb+0x20): undefined reference to `devilution::SetRndSeed(unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj: in function `_GLOBAL__sub_D__ZN10devilution23gbSomebodyWonGameKludgeE':
Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution7NetInitEb+0x600): undefined reference to `devilution::SetRndSeed(unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution7NetInitEb+0x2ac): undefined reference to `devilution::nthread_send_and_recv_turn(unsigned int, int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution7NetInitEb+0x550): undefined reference to `devilution::nthread_send_and_recv_turn(unsigned int, int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution7NetInitEb+0x1b4): undefined reference to `devilution::SNetInitializeProvider(unsigned int, devilution::GameData*)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj: in function `_GLOBAL__sub_D__ZN10devilution23gbSomebodyWonGameKludgeE':
Source/CMakeFiles/libdevilutionx.dir/multi.cpp.obj:(.text._ZN10devilution7NetInitEb+0x73a): undefined reference to `devilution::SNetInitializeProvider(unsigned int, devilution::GameData*)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/pfile.cpp.obj:(.text._ZN10devilution15OpenSaveArchiveEj+0x26): undefined reference to `devilution::MpqArchive::Open(char const*, int&)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/pfile.cpp.obj:(.text._ZN10devilution16OpenStashArchiveEv+0x1e): undefined reference to `devilution::MpqArchive::Open(char const*, int&)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/pfile.cpp.obj:(.text._ZN10devilution11ReadArchiveERNS_10MpqArchiveEPKcPm+0x1e): undefined reference to `devilution::MpqArchive::ReadFile(char const*, unsigned long&, int&)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/pfile.cpp.obj:(.text._ZN10devilution23pfile_ui_set_hero_infosEPFbPNS_11_uiheroinfoEE+0x66): undefined reference to `devilution::MpqArchive::Open(char const*, int&)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/pfile.cpp.obj:(.text._ZN10devilution27pfile_read_player_from_saveEjRNS_6PlayerE+0x78): undefined reference to `devilution::MpqArchive::Open(char const*, int&)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/pfile.cpp.obj:(.text._ZN10devilution23pfile_compare_hero_demoEib+0x612): undefined reference to `devilution::MpqArchive::Open(char const*, int&)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/engine/demomode.cpp.obj:(.text._ZN10devilution4demo13RecordMessageERK9SDL_Eventt+0x124): undefined reference to `devilution::IsCustomEvent(unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/engine/demomode.cpp.obj:(.text._ZN10devilution4demo13RecordMessageERK9SDL_Eventt+0x138): undefined reference to `devilution::GetCustomEvent(unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: /opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/libm020/libm881/libstdc++.a(c++locale.o):(.text._ZSt14__convert_to_vIeEvPKcRT_RSt12_Ios_IostateRKPi+0x28): undefined reference to `strtold'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/devilutionx.dir/build.make:424: devilutionx] Error 1
make[1]: *** [CMakeFiles/Makefile2:358: CMakeFiles/devilutionx.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

Exited with code exit status 2


```

**Comment by AJenbo** [comment]
Still getting a few linker issues.
```
_deps/libfmt-build/libfmt.a(format.cc.obj): duplicate section `.text._ZN3fmt2v96detail15write_codepointILm2EcNS0_8appenderEEET1_S4_cm' has different contents
_deps/libfmt-build/libfmt.a(format.cc.obj): duplicate section `.text._ZN3fmt2v96detail15write_codepointILm2EcNS0_8appenderEEET1_S4_cm' has different contents
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/init.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_17LoadMPQERKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS7_EESt17basic_string_viewIcS5_E+0x1cc): undefined reference to `devilution::MpqArchive::Open(char const*, int&)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/init.cpp.obj: in function `_GLOBAL__sub_D__ZN10devilution8gbActiveE':
Source/CMakeFiles/libdevilutionx.dir/init.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_17LoadMPQERKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS7_EESt17basic_string_viewIcS5_E+0x2f8): undefined reference to `devilution::MpqArchive::ErrorMessage(int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/missiles.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_121UpdateMissileVelocityERNS_7MissileENS_7PointOfIiEEi+0x4e): undefined reference to `hypot'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/missiles.cpp.obj: in function `_GLOBAL__sub_D__ZN10devilution8MissilesB5cxx11E':
Source/CMakeFiles/libdevilutionx.dir/missiles.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_121UpdateMissileVelocityERNS_7MissileENS_7PointOfIiEEi+0xe8): undefined reference to `hypot'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/engine/demomode.cpp.obj:(.text._ZN10devilution4demo13RecordMessageERK9SDL_Eventt+0x124): undefined reference to `devilution::IsCustomEvent(unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/engine/demomode.cpp.obj:(.text._ZN10devilution4demo13RecordMessageERK9SDL_Eventt+0x138): undefined reference to `devilution::GetCustomEvent(unsigned int)'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: /opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/libm020/libm881/libstdc++.a(c++locale.o):(.text._ZSt14__convert_to_vIeEvPKcRT_RSt12_Ios_IostateRKPi+0x28): undefined reference to `strtold'
```

**Comment by AJenbo** [comment]
Update
```
_deps/libfmt-build/libfmt.a(format.cc.obj): duplicate section `.text._ZN3fmt2v96detail15write_codepointILm2EcNS0_8appenderEEET1_S4_cm' has different contents
_deps/libfmt-build/libfmt.a(format.cc.obj): duplicate section `.text._ZN3fmt2v96detail15write_codepointILm2EcNS0_8appenderEEET1_S4_cm' has different contents
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/missiles.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_121UpdateMissileVelocityERNS_7MissileENS_7PointOfIiEEi+0x4e): undefined reference to `hypot'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: Source/CMakeFiles/libdevilutionx.dir/missiles.cpp.obj: in function `_GLOBAL__sub_D__ZN10devilution8MissilesB5cxx11E':
Source/CMakeFiles/libdevilutionx.dir/missiles.cpp.obj:(.text._ZN10devilution12_GLOBAL__N_121UpdateMissileVelocityERNS_7MissileENS_7PointOfIiEEi+0xe8): undefined reference to `hypot'
/opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/../../../../m68k-amigaos/bin/ld: /opt/m68k-amigaos-gcc10/lib/gcc/m68k-amigaos/13.1.1/libm020/libm881/libstdc++.a(c++locale.o):(.text._ZSt14__convert_to_vIeEvPKcRT_RSt12_Ios_IostateRKPi+0x28): undefined reference to `strtold'

```

**Comment by AJenbo** [comment]
good idea to make it a debug build for now

**Comment by StephenCWills** [comment]
When I removed `-O2`, I received a lot of linker errors.

Putting `-O2` back, but leaving `-DCMAKE_BUILD_TYPE=Debug` results in a binary that is no longer executable, according to @arczi84.

**Comment by AJenbo** [comment]
Application launches but fails to create a file during the read/write test:

image

Maybe an issue with libc?

File: Packaging/amiga/prep.sh
Comment (arczi84) [inline Packaging/amiga/prep.sh:1]: gcc10?

Comment (AJenbo) [reply Packaging/amiga/prep.sh:1]: that's the name of the branch for GCC 13.1, not something we control

File: .circleci/config.yml [hunk 2]
```diff
@@ -30,12 +30,12 @@ jobs:
...
   amigaos-m68k:
     docker:
-      - image: amigadev/crosstools:m68k-amigaos
+      - image: amigadev/crosstools:m68k-amigaos-gcc10
     working_directory: ~/repo
     steps:
       - checkout
       - run: Packaging/amiga/prep.sh
-      - run: cmake -S. -Bbuild -DM68K_CPU=68040 -DM68K_FPU=hard -DM68K_COMMON="-s -fbbb=- -ffast-math -O2"
+      - run: cmake -S. -Bbuild -DCMAKE_BUILD_TYPE=Debug -DM68K_CPU=68040 -DM68K_FPU=hard -DM68K_COMMON="-s -fbbb=- -ffast-math -O2"
```
Comment (AJenbo) [inline .circleci/config.yml:38]:
```suggestion
      - run: cmake -S. -Bbuild -DM68K_CPU=68040 -DM68K_FPU=hard -DM68K_COMMON="-s -fbbb=- -ffast-math -O2"
```

File: Packaging/amiga/prep.sh [hunk 3]
```diff
@@ -15,6 +15,8 @@ export M68K_CXXFLAGS="${M68K_CPU_FPU} ${M68K_COMMON}"
 
 PARALLELISM="$(getconf _NPROCESSORS_ONLN)"
 
+ln -s /opt/m68k-amigaos-gcc10 /opt/m68k-amigaos
+
```
Comment (AJenbo) [inline Packaging/amiga/prep.sh:19]:
```suggestion
```

**Pull request merged**